shared/lib/canSafelyAutoCloseThisPopup.ts

```diff
@@ -0,0 +1,28 @@
+import browser from 'webextension-polyfill';
+/**
+ * Determines if it's safe to automatically close the current window.
+ *
+ * This function checks if the current window is a popup. The function follows
+ * a "safe by default" approach - it only prevents auto-closing when it can
+ * definitively confirm that the window is not a popup (to avoid closing
+ * extension UIs running in tabs).
+ *
+ * @returns A Promise that resolves to `true` if the window is a popup
+ * or if the window type cannot be determined – and `false` only if the
+ * window is confirmed to not be a popup.
+ */
+export const canSafelyAutoCloseThisPopup = async (): Promise<boolean> => {
+  try {
+    const currentWindow = await browser.windows.getCurrent();
+    // Only return false if we know for sure it's not a popup
+    if (currentWindow?.type && currentWindow.type !== 'popup') {
```

Comment:
oh wow, it's so simple! how did we miss this for so long?

Reply:
According to the return type of https://developer.chrome.com/docs/extensions/reference/api/windows#method-getCurrent this method will always return a window, so we shouldn't need to `?` check first. The "current" window is the window running the code that is calling `browser.windows.getCurrent()`, so it must always exist, right? Are the types saying claiming something different?

Reply:
nit: I noticed there is a `browser.windows.WindowType.POPUP` property thath returns the "popup" string.

Reply:
Additionally, I just learned of `browser.tabs.getCurrent()`. Do you know of any reason we can't use this to close only the _`tab`_ and not the `window` when we detect that the window is not a `"popup"`?

```javascript
const tab = await browser.tabs.getCurrent();
await browser.tabs.remove(tab.id);
```

Reply:
I added that to the call in actions.ts.